Assets/Scripts/Models/Area/RoomManager.cs

```diff
@@ -0,0 +1,427 @@
+
+            // Add the outside rooom to the list just in case.
+            rooms.Add(OutsideRoom);
+        }
+
+        #region Events
+
+        /// <summary>
+        /// Occurs when adding a new room to the manager, 
+        /// before the room is actually added.
+        /// </summary>
+        public event Action<Room> Adding;
+
+        /// <summary>
+        /// Occurs after the room has been added to the manager.
+        /// </summary>
+        public event Action<Room> Added;
+
+        /// <summary>
+        /// Occurs when a room is split.
+        /// The first room is the old one, 
+        /// the second is the new one.
+        /// </summary>
+        public event Action<Room, Room> Split;
+
+        /// <summary>
+        /// Occurs when two rooms are joined.
+        /// The first room is the old one, 
+        /// the second is the new one.
+        /// 
+        /// Warning, the old room has already been removed from
+        /// the manager, and most of its data can be considered
+        /// stale. This is a possible FIXME.
+        /// </summary>
+        public event Action<Room, Room> Joined;
+
+        /// <summary>
+        /// Occurs when removing a room from the manager, 
+        /// before the room is actually removed.
+        /// </summary>
+        public event Action<Room> Removing;
+
+        /// <summary>
+        /// Occurs after the room has been removed 
+        /// from the manager.
+        /// </summary>
+        public event Action<Room> Removed;
+
+        #endregion
+
+        /// <summary>
+        /// Gets the amount of rooms.
+        /// </summary>
+        /// <value>The number of rooms managed.</value>
+        public int Count
+        {
+            get
+            { 
+                return rooms.Count;
+            }
+        }
+
+        /// <summary>
+        /// Gets the outside room.
+        /// </summary>
+        /// <value>The outside room.</value>
+        public Room OutsideRoom { get; private set; }
+
+        /// <summary>
+        /// Gets a managed room from an ID.
+        /// </summary>
+        /// <param name="index">The ID of the room.</param>
+        public Room this [int index]
+        {
+            get
+            { 
+                if (index < 0 || index > Count - 1)
+                {
+                    return null;
+                }
+
+                return rooms[index];
+            }
+        }
+
+        /// <summary>
+        /// Gets the room ID.
+        /// </summary>
+        /// <returns>The room ID.</returns>
+        /// <param name="room">What Room to check for ID.</param>
+        public int GetRoomID(Room room)
+        {
+            return rooms.IndexOf(room);
+        }
+
+        /// <summary>
+        /// Add a room to the manager.
+        /// </summary>
+        /// <param name="room">A room to be managed.</param>
+        public void Add(Room room)
+        {
+            World.Current.roomGraph = null;
```

Comment:
shouldn't this be the world registering to the adding & removing event? this is not the responsability of the room manager